tests/Sidebar.test.js

```diff
@@ -0,0 +1,191 @@
+  navLinks,
+} = sidebarProps;
+
+describe("Sidebar", () => {
+  it("should render without error", () => {
+    const { getByText } = render(
+      <Router>
+        <Sidebar {...sidebarProps} />
+      </Router>
+    );
+    expect(getByText(subdomain)).toBeInTheDocument();
+  });
+
+  it("should not display text when collapsed", () => {
+    const { queryByText } = render(
+      <Router>
+        <Sidebar {...sidebarProps} isCollapsed />
+      </Router>
+    );
+    expect(queryByText(subdomain)).not.toBeInTheDocument();
+  });
+
+  it("should display organization name correctly", () => {
+    const { getByText } = render(
+      <Router>
+        <Sidebar {...sidebarProps} />
+      </Router>
+    );
+    expect(getByText(orgName)).toBeInTheDocument();
+  });
+
+  it("should display all navlink elements correctly", () => {
+    const { getByText } = render(
+      <Router>
+        <Sidebar {...sidebarProps} />
+      </Router>
+    );
+    navLinks.forEach(({ label, to }) => {
+      const sideNavLabel = getByText(label);
+      const sideNavLink = sideNavLabel.parentElement;
+      expect(getByText(label)).toBeInTheDocument();
+      expect(sideNavLink["href"].endsWith(to)).toBe(true);
+    });
+  });
+
+  it("should display profile information correctly", async () => {
+    const { getByText, container, queryByText } = render(
+      <Router>
+        <Sidebar {...sidebarProps} isCollapsed />
+      </Router>
+    );
+
+    const profileImage = container.querySelector(`img[src="${imageUrl}"]`);
+    expect(profileImage).toBeInTheDocument();
+
+    expect(queryByText(email)).not.toBeInTheDocument();
+
+    userEvent.hover(profileImage);
+    await waitFor(() => getByText(email));
+    expect(getByText(userName)).toBeInTheDocument();
+    expect(getByText(bottomLinks[0].label)).toBeInTheDocument();
+  });
+
+  it("should apply featured toolTipStyle correctly", async () => {
+    const { getByText, container } = render(
+      <Router>
+        <Sidebar {...sidebarProps} isCollapsed tooltipStyle="featured" />
+      </Router>
+    );
+    const { to, label, description } = navLinks[0];
+    const navLink = container.querySelector(`a[href="${to}"]`);
+    expect(navLink).toBeInTheDocument();
+
+    userEvent.hover(navLink);
+    expect(await waitFor(() => getByText(label))).toBeInTheDocument();
+    expect(await waitFor(() => getByText(description))).toBeInTheDocument();
+  });
+
+  it("should apply default toolTipStyle correctly", async () => {
+    const { queryByText, getByText, container } = render(
+      <Router>
+        <Sidebar {...sidebarProps} isCollapsed tooltipStyle="default" />
+      </Router>
+    );
+
+    for (const link of navLinks) {
+      const { to, label, description } = link;
+      const navLink = container.querySelector(`a[href="${to}"]`);
+      expect(navLink).toBeInTheDocument();
+
+      userEvent.hover(navLink);
+      expect(await waitFor(() => getByText(label))).toBeInTheDocument();
```

Comment:
Here instead of `await waitFor(() => getByText(label))`, could you try `await findByText(label)`